fc-community-extension/src/utils/backend.js

```diff
@@ -24,6 +25,37 @@ export const getNotes = async (queryparams) => {
   }
 };
 
+export const getWarpcastNotes = async (queryparams) => {
+  const notes = await getNotes(queryparams, 'BASE_MAINNET');
+  // Warpcasts users manually enter the post URL on the frame
+  // TODO: validate input inside the frame
+  // TODO: filter invalid notes in the backend
+  const validNotes = notes
+    .map((note) => {
+      try {
+        new URL(note.postUrl);
+        return note;
+      } catch (error) {
+        console.error(`Invalid post URL: ${note.postUrl}`);
+        return null;
+      }
+    })
+    .filter((url) => url !== null);
+  return validNotes;
```

Comment:
This is true for any network really since I can just push transactions manually to the contract.

Let's
1. move this validation to `getNotes`,
2. have a list of networks supported in `constant.js` with both network name and chainid, 
3. change `getNotesForAllSocials` to be a `map` over the valid networks + call `getNotes` for each of them